### PR TITLE
Add support for using image digests instead of tags

### PR DESCRIPTION
**Pull request by ryantking**
**Description of the change:**

Adds the ability to create operators that use image digests instead of tags for operator images as well as related images. 

**Motivation for the change:**

https://github.com/operator-framework/enhancements/blob/master/enhancements/sha-bundle-digest.md

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: changelog/fragments/image-digests.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+entries:
+  - description: >
+      Adds support to bundle operators using image digests instead of tags.
+    kind: "addition"
+    breaking: false
```
Comment (camilamacedo86) [inline changelog/fragments/image-digests.yaml:5]: We are changing the scaffold. So, we need to provide the migration steps for authors to know how they need to update their makefile to have it.

**Comment by jmrodri** [comment]
@ryantking I forced an update to this branch so I could add the Signoff for Ish's initial commit which was causing DCO to fail. And I added the migration for camila.

File: changelog/fragments/image-digests.yaml (hunk 1 above)
Comment (jmrodri) [reply changelog/fragments/image-digests.yaml:5]: Done https://github.com/operator-framework/operator-sdk/pull/5567/commits/80d7022de4de649c713eb18f7a549ee3009d8375

**Review by jmrodri — APPROVED** [review]
/lgtm

**Pull request merged**